domain/mocks/pricing_update_listener_mock.go

```diff
@@ -4,12 +4,13 @@ import (
 	"context"
 	"time"
 
+	"github.com/osmosis-labs/osmosis/osmomath"
 	"github.com/osmosis-labs/sqs/domain"
 )
 
 type PricingUpdateListenerMock struct {
 	Height                 int
-	PricesBaseQuteDenomMap map[string]map[string]any
+	PricesBaseQuteDenomMap map[string]map[string]osmomath.BigDec
```

Comment:
Just question not comment - guess we have to use osmomath.BigDec as it involves price precision, right? I remember you said excessive use of osmomath.BigDec can have performance impact

Reply:
Yes, it does have a performance impact since it is 36 decimals as opposed to 18 decimals with regular Dec.

We allocate raw bytes to represent this structure under the hood so the space complexity is proportional to the difference. The performance cost is most notable in intermediary math operations since we know need to operate on more bytes.

The native ints are usually preferred for performance, if they are sufficient to represent what we need. 

With the on-chain price, however, we need 36 decimals to reduce the likelihood of truncations in intermediary operations since prices can either be large or small due to a potential for orders of magnitude in difference in reserves. Additionally, decimal precision exponents exacerbate this. As a result, using BigDec` for prices at the expense of performance is an acceptable performance cost.

This is also not the performance bottleneck today